ui/app/sample-data/demoDashboard.ts

```diff
@@ -27,20 +27,32 @@ const demoDashboard: DashboardResource = {
     duration: '30m',
     variables: [
       {
-        kind: 'TextVariable',
+        kind: 'ListVariable',
         spec: {
           name: 'job',
-          value: 'node',
+          plugin: {
```

Comment:
I think the big question is here, should each variable type be its own variable plugin? So instead of:
```ts
{
   kind: "PrometheusVariable",
   spec: {
      datasource: "prom-demo",
      kind: "LabelValues",
      spec: { 
         label: "job"
      }
   }
}
```

It would look something like this:

```ts
{
   kind: "PrometheusLabelValuesVariable",
   spec: {
      datasource: "prom-demo",
      label: "job"
   }
}
```

I have a slight preference for the first approach (basically overloading the Prom Variable plugin) and treating the types of queries as properties in the spec. That way, they can be visually grouped together and the datasource selection is preserved.

Reply:
The first one feels best to me too!

Reply:
Yeah I like the first version as well, seems more consistent with Luke's recent changes too. Would there need to be a `plugin` added too though (similar to [ex here](https://github.com/perses/perses/blob/main/ui/app/sample-data/benchmarkDashboard.ts#L114))?

Reply:
I'm not sure if it makes sense to include `plugin` since it's the plugin kind, the `PrometheusVariable` is the actual plugin kind. If that's a bit unclear, we can also just change `kind` to `type` in the sub config to differentiate between an entity definition.

```ts
{
   kind: "PrometheusVariable",
   spec: {
      datasource: "prom-demo",
      type: "LabelValues", // or LabelNames
      query: { 
         label: "job"
      }
   }
}
```

Reply:
We went with the one plugin per query type approach as discussed in the sync.